src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,77 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieCard } from '../MovieCard';
+
+type Props = {
+  movies: Movie[];
+  setMovies: (value: Movie[]) => void;
+};
+
+export const FindMovie: React.FC<Props> = ({
+  movies, setMovies,
+}) => {
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [title, setTitle] = useState('');
+  const [loading, setLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setIsError(false);
+  };
+
+  const handleAddClick = () => {
+    if (movie && !movies.some(m => m.imdbId === movie.imdbId)) {
+      setMovies([...movies, movie]);
+    }
+
+    setMovie(null);
+    setTitle('');
+  };
+
+  const handleFormSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setLoading(true);
+
+    getMovie(title)
+      .then((response) => {
+        if ('Error' in response) {
```

Comment:
Consider throwing an error instead of setting the state and returning. This way, you can catch the error in the catch block.

Reply:
you must catch error in catch block and there you can set isError